core/offchain/src/api.rs

```diff
@@ -188,7 +190,19 @@ impl<Storage, KeyProvider> OffchainExt for Api<Storage, KeyProvider> where
 	}
 
 	fn timestamp(&mut self) -> Timestamp {
-		unavailable_yet("timestamp")
+		let now = SystemTime::now();
+		let epoch_duration = now.duration_since(SystemTime::UNIX_EPOCH);
+		match epoch_duration {
+			Err(_) => {
+				warn!("Current time is earlier than UNIX_EPOCH.");
+				Timestamp::from_unix_millis(0)
+				},
+			Ok(d) => {
+				let duration = d.as_millis();
+				// Assuming overflow won't happen for a few hundred years.
```

Comment:
It's actually billions of years :boom: This comment should end up as an `expect`